acme/http_challenge.go

```diff
@@ -95,12 +95,13 @@ Loop:
 func (s *httpChallenge) startHTTPServer(domain string, token string, keyAuth string) {
 
 	// Allow for CLI port override
-	port := ":80"
+	port := domain + ":80"
 	if s.optPort != "" {
 		port = ":" + s.optPort
 	}
```

Comment:
This PR seems to ignore or negate the fallback (see below, L107)... I don't think this will work.